controllers/library.py

```diff
@@ -0,0 +1,112 @@
+from mongoengine import DoesNotExist
+from utils.custom_errors import *
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+"""
+Represents a library system with various attributes for managing its system.
+
+Attributes:
+    library_items (Dict[str, LibraryItem]): Dictionary of library items.
+    patrons (Dict[str, Patron]): Dictionary of library patrons.
+    bills (Dict[str, float]): Dictionary of patron bills.
+    name (str): The name of the library.
+"""
+
+
+class Library(BaseModel):
+    library_items: Dict[str, LibraryItem] = {}
+    patrons: Dict[str, Patron] = {}
+    bills: Dict[str, float] = {}  # Dictionary to keep track of all bills, keyed by patron ID
+    name: str
+
+    def add_new_library_items_to_the_library(self, new_library_items: List[LibraryItem]):
+        """
+        Add new library items to the library system.
+
+        Args:
+            new_library_items (List[LibraryItem]): List of library items to add.
+        """
+        for new_library_item in new_library_items:
+            if new_library_item.isbn in self.library_items.keys():
+                raise NotUniqueLibraryItemError(
+                    f"The library item with ISBN {new_library_item.isbn} already exists in the library"
+                )
+            self.library_items[new_library_item.isbn] = new_library_item
+            new_library_item_document = item_pydantic_to_mongoengine(new_library_item)
+            insert_document_dal('library-items', new_library_item_document)
+            logging.info(
+                f"The library item {new_library_item.title} with ISBN {new_library_item.isbn} was added to '{self.name}' library"
+            )
+
+    def add_new_patron_to_the_library(self, patrons_to_add: List[Patron]):
+        """
+        Add new patrons to the library system.
+
+        Args:
+            patrons_to_add (List[Patron]): List of patrons to add.
+        """
+        for patron in patrons_to_add:
+            if patron.patron_id in self.patrons:
+                raise NotUniquePatronError(f"The patron with ID {patron.patron_id} already exists in the library")
+            self.patrons[patron.patron_id] = patron
+            patron_document = patron_pydantic_to_mongoengine(patron)
+            insert_document_dal('library-patrons', patron_document)
+            logging.info(f"The patron {patron.name} with ID {patron.patron_id} was added to the library")
+
+    def remove_patrons_from_the_library(self, patrons_to_remove: List[Patron]):
+        """
+        Remove patrons from the library system.
+
+        Args:
+            patrons_to_remove (List[Patron]): List of patrons to remove.
+        """
+
+        for patron in patrons_to_remove:
+            if patron.patron_id not in self.patrons:
+                raise NotExistingPatron(f"The patron with ID {patron.patron_id} does not exist in the library")
```

Comment:
You need to rename your errors to be in pas tense. For example: PatronNotFound or PatronAlreadyExists.
The same comments about the other errors also apply here